src/main/java/ch/njol/skript/expressions/ExprSortedList.java

```diff
@@ -48,13 +49,32 @@
 	}
 	
 	@SuppressWarnings("null")
-	private Expression<Object> list;
+	private Expression<?> list;
+	
+	@Override
+	public Class<? extends Object> getReturnType() {
+		return list.getReturnType();
+	}
+	
+	@Override
+	public boolean isSingle() {
+		return false;
+	}
 	
 	@SuppressWarnings({"null", "unchecked"})
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		Class<? extends Object> type = exprs[0].getReturnType();
-		if (Comparable.class.isAssignableFrom(type)) {
+		if (exprs[0] instanceof Variable<?>) {
+			Variable<?> variable = ((Variable<?>) exprs[0]);
+			if (!variable.isList()) {
+				Skript.error("You may only sort list variables, not single variables.");
+				return false;
+			}
+			list = exprs[0].getConvertedExpression(Object.class);
+			return true;
+		}
+		if (!Comparable.class.isAssignableFrom(type)) {

```

Comment:
Reverse list **does not** need sorting (Comparable) to be implemented in the elements.